mdata/cache/ccache_metric.go

```diff
@@ -116,12 +231,26 @@ func (mc *CCacheMetric) Add(prev uint32, itergen chunk.IterGen) {
 		}
 	}
 
-	// regenerate the list of sorted keys after adding a chunk
-	mc.generateKeys()
+	mc.addKey(ts)
 
 	return
 }
 
+func (mc *CCacheMetric) addKey(ts uint32) {
```

Comment:
this function has only one call-site, so it might as well be integrated there to save a function call

Reply:
can argue both ways on this. function is small, compiler would probably inline it, and overhead is minimal.  it's kind of nice to keep it in separate function to keep cyclomatic complexity low.

OTOH it arguably simplifies the code a bit, so i'm moving it as you suggest.